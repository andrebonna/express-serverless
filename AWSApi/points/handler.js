'use strict';

module.exports.create = (event, context, callback) => {
    const response = {
        statusCode: 200,
        body: JSON.stringify({
            message: event
        })
    };
    callback(null, response);
};
