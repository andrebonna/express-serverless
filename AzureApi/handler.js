'use strict';

/* eslint-disable no-param-reassign */

module.exports.hello = function (context) {
  context.log('JavaScript HTTP trigger function processed a request.');

  context.res = {
    // status: 200, /* Defaults to 200 */
    body: JSON.stringify(context),
  };

  context.done();
};
