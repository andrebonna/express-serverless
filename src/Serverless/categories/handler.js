import React from 'react';
import templateBuilder from '../commons/templateBuilder';
import constants from '../commons/constants';
import Categories from '../../View/categories/Categories';

export function get(event, context, callback) {
    templateBuilder({
        title: 'Categories', 
        metas: constants.metas,
        props: {
            children: <Categories />
        }
    }, (err, data) => {
        const response = {
            statusCode: 200,
            headers: {
                'Content-type': 'text/html'
            },
            body: data
        };
        callback(null, response);
    });
}
