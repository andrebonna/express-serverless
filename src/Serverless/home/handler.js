import React from 'react';
import AWS from 'aws-sdk';
import templateBuilder from '../commons/templateBuilder';
import constants from '../commons/constants';
import Home from '../../View/home/Home';

const dynamodb = new AWS.DynamoDB.DocumentClient({ region: 'us-east-1' });

function getImagesFiltered(queryStringParameters) {
    let params = {
        TableName: 'Photos',
        Select: 'ALL_ATTRIBUTES'
    };

    if (queryStringParameters && queryStringParameters.filter) {
        params = {
            ...params,
            ScanFilter: {
                Categories: {
                    ComparisonOperator: 'CONTAINS',
                    AttributeValueList: [queryStringParameters.filter]
                }
            }
        };
    }

    return dynamodb.scan(params).promise();
}

function getCategories(data) {
    return dynamodb.scan({
        TableName: 'Photos',
        Select: 'SPECIFIC_ATTRIBUTES',
        AttributesToGet: ['Categories']
    }).promise().then((categories)=>{
        const categoriesSet = new Set();
        categories.Items.map(({ Categories })=>{
            Categories.forEach(category => categoriesSet.add(category));
        });          
        return { 
            data, 
            categories: [...categoriesSet] 
        };
    });
}

export function get(event, context, callback) {

    const {
        queryStringParameters
    } = event;

    getImagesFiltered(queryStringParameters)
        .then(getCategories)
        .then(({data, categories}) => {
            templateBuilder({
                title: 'Andr√©a Buck Photos',
                metas: constants.metas,
                props: {
                    categories,
                    children: <Home images={data.Items} />
                }
            }, (err, data) => {
                if (err) {
                    callback(err);
                }
                const response = {
                    statusCode: 200,
                    headers: {
                        'Content-type': 'text/html'
                    },
                    body: data
                };
                callback(null, response);
            });
        }).catch(callback);
}
