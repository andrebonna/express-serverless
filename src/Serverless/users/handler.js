import templateBuilder from '../commons/templateBuilder';

export function get(event, context, callback) {
    templateBuilder({
        title: 'Serverless Universal React', 
        metas: event.queryStringParameters
    }, (err, data) => {
        const response = {
            statusCode: 200,
            headers: {
                'Content-type': 'text/html'
            },
            body: data
        };
        callback(null, response);
    });
}

export function create(event, context, callback) {
    const response = {
        statusCode: 200,
        body: event.body
    };
    callback(null, response);
}
