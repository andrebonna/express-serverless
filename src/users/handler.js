'use strict';

module.exports.create = (event, context, callback) => {
    const response = {
        statusCode: 200,
        body: event.body
    };
    callback(null, response);
};
