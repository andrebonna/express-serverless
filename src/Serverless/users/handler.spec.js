import { get, create } from './handler';

describe('Handler Lambdas', ()=>{
    it('Test Get Users Page', (done) => {
        get({}, null, (_, response)=>{
            expect(response.body).toMatchSnapshot();
            done();
        });
    });

    it('Test Create Users', (done) => {
        create({
            body: 'bodyMock'
        }, null, (_, response)=>{
            expect(response.body).toMatchSnapshot();
            done();
        });
    });
});
