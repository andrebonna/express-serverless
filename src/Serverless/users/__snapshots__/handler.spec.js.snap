exports[`Handler Lambdas Test Create Users 1`] = `"bodyMock"`;

exports[`Handler Lambdas Test Get Users Page 1`] = `
"<!DOCTYPE html>
<html lang=\"en\">
    <head data-reactroot=\"\" data-reactid=\"1\" data-react-checksum=\"-1418386197\"><title data-reactid=\"2\">Serverless Universal React</title></head>
    <body class=\"container\">
        <div id=\"content\"><div data-reactroot=\"\" data-reactid=\"1\" data-react-checksum=\"-1012498656\"><header data-reactid=\"2\"><h1 data-reactid=\"3\">Hello World App</h1></header><section data-reactid=\"4\"><p data-reactid=\"5\">Hello world!</p><button data-reactid=\"6\"><!-- react-text: 7 -->Count: <!-- /react-text --><!-- react-text: 8 -->0<!-- /react-text --></button></section><footer data-reactid=\"9\"><p data-reactid=\"10\">© André Bonna</p></footer></div></div>
        <script src=\"bundle.js\"></script>
    </body>
</html>"
`;
